src/api.c

```diff
@@ -117,7 +117,12 @@ yaml_string_join(
 YAML_DECLARE(int)
 yaml_stack_extend(void **start, void **top, void **end)
 {
-    void *new_start = yaml_realloc(*start, ((char *)*end - (char *)*start)*2);
```

Comment:
Are we sticking to C89? What are our supported operating systems if we're deciding to stick to that?